### ISSUE TITLE
extractUASTs UDF sends requests to Babelfish repeatedly even if there is no driver for the given language

### ISSUE THREAD

**Issue by smola** [src-d/engine#355]
## Expected Behavior
`extractUASTs` UDF should be able to avoid unnecessary Babelfish requests if the Babelfish server does not support the language.

## Current Behavior
Babelfish requests are always sent even if language is not supported.

Example log:
```
18/04/04 09:40:02 WARN Bblfsh: FATAL server_ui/media/main.css: unexpected error: runtime failure: missing driver for language "css"
18/04/04 09:40:02 WARN Bblfsh: FATAL server_ui/templates/about.html: unexpected error: runtime failure: missing driver for language "html"
18/04/04 09:40:03 WARN Bblfsh: FATAL server_ui/templates/base.html: unexpected error: runtime failure: missing driver for language "html"
18/04/04 09:40:03 WARN Bblfsh: FATAL server_ui/templates/confirm.html: unexpected error: runtime failure: missing driver for language "html"
18/04/04 09:40:03 WARN Bblfsh: FATAL server_ui/templates/docs.html: unexpected error: runtime failure: missing driver for language "html"
18/04/04 09:40:03 WARN Bblfsh: FATAL server_ui/templates/done.html: unexpected error: runtime failure: missing driver for language "html"
```

## Possible Solution
1. On first Babelfish request, query the Babelfish server for all supported languages, then cache it.
2. Or, on Babelfish error, if the server returns an error about not supported language / no driver, cache it.

I would be inclined to to the first one. We might need, however, a way to purge this cache.

## Steps to Reproduce (for bugs)
1. Perform a query with extractUASTs on `html` or `css`.
2. See the logs (example above).

## Context
This is not blocking me and I don't really know the relative performance impact of this.

## Your Environment
* Version used: 0.5.5 (using engine-jupyter)
* Operating System and version: Ubuntu Xenial
* Some needed resources to reproduce the problem:

**Comment by erizocosmico** [comment]
AFAIK, the Scala client does not provide any way to ask the server if the language is supported or not. If we want that, it'd have to be with a manually maintained list of languages.

**Comment by ajnavarro** [comment]
@smola @erizocosmico maybe we should do a feature request to bblfsh team?

**Comment by smola** [comment]
@ajnavarro @erizocosmico Yes. Let's wait for the Scala client to provide that functionality: https://github.com/bblfsh/client-scala/issues/68

**Comment by bzz** [comment]
Just FYI, https://github.com/bblfsh/client-scala/pull/69 was merged and `supportedLanguages` method is available now in scala-client as of [v1.9.1](https://github.com/bblfsh/client-scala/releases/tag/v1.9.1)
